zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/factory/SharedZkClientFactory.java

```diff
@@ -125,4 +132,74 @@ public int getActiveConnectionCount() {
     }
     return count;
   }
+
+  public interface OnCloseCallback {
+    /**
+     * Triggered after the SharedZkClient is closed.
+     */
+    void onClose();
+  }
+
+  /**
+   * NOTE: do NOT use this class directly. Please use SharedZkClientFactory to create an instance of SharedZkClient.
+   * InnerSharedZkClient is a ZkClient used by SharedZkClient to power ZK operations against a single ZK realm.
+   */
+  public static class InnerSharedZkClient extends ZkClient implements HelixZkClient {
```

Comment:
So basically InnerSharedZkClient is exactly the original SharedZkClient. Is the code exactly the same?

Reply:
This is actually what I worried, the code here changed. How do we reason that the new implementation of InnerSharedZkClient keeps the old InnerSharedZkClient original behavior? 
@jiajunwang can you have a look?

Reply:
@kaisun2000 
InnerSharedZkClient is meant to preserve the behavior/logic of the original SharedZkClient.

1. Clearly the code is not "exactly" the same. You have access to the original SharedZkClient - please do a comparison.
2. What is the exact concern here? I'm not exactly sure what your question is.

Reply:
I had a careful look, this seems to be OK to maintain the original behavior.  Assertion 1.